mmvb_backend/case_synthesizer/generator.py

```diff
@@ -15,45 +16,173 @@
+        symptoms.append(
+            {
+                "id": symptom["id"],
+                "name": symptom["term"],
+                "standardOntologyUris": [symptom["sctid"]],
+                "state": state,
+                "attributes": attributes,
+            }
+        )
+    return presenting_symptom, symptoms
+
+
+def generate_case_data():
+    age = random.randint(MIN_AGE, MAX_AGE)
+    biological_sex = random.choice(BIOLOGICAL_SEXES)
+
+    sampled_factor_ids = [
+        factor["id"]
+        for factor in FIXTURES_DATA["factors"]
+        if random.random() < FACTOR_PROBABILITY
+    ]
+
+    # Sample a condition based on priors and factors
+    condition_priors = []
+    for condition in FIXTURES_DATA["conditions"]:
+        prior = condition["prior"]
+        sex_factor = next(
+            item["weights"][biological_sex]
+            for item in FIXTURES_DATA["condition_sex_weights"]
+            if item["condition_id"] == condition["id"]
+        )
+        prior *= sex_factor
+        for factor in FIXTURES_DATA["condition_factor_weights"]:
+            if factor["condition_id"] == condition["id"]:
+                if factor["factor_id"] in sampled_factor_ids:
+                    prior *= factor["weight"]
+        condition_priors.append(prior)
+
+    sampled_condition = random.choices(
+        FIXTURES_DATA["conditions"], condition_priors
+    )[0]
+
+    # Sample symptoms based on weight of connection to sampled condition
+    related_symptom_weights = [
+        weight_info
+        for weight_info in FIXTURES_DATA["condition_symptom_weights"]
+        if weight_info["condition_id"] == sampled_condition["id"]
+    ]
+    presenting_symptom, symptoms = sample_symptoms(related_symptom_weights)
+
+    if len(presenting_symptom) < 1:
+        raise SynthesisError(
+            "Something went wrong when sampling symptom states"
+        )
 
-    return {
-        symptom: state for symptom, state in symptom_states.items() if state
+    case_data = {
+        "caseData": {
+            "profileInformation": {
+                "age": age,
+                "biologicalSex": biological_sex,
+            },
+            "presentingComplaints": [presenting_symptom],
+            "otherFeatures": symptoms,  # ADD FACTORS
+        },
+        "valuesToPredict": {
+            "expectedTriageLevel": sampled_condition["triage"],
+            "expectedCondition": {
+                "id": sampled_condition["id"],
+                "name": sampled_condition["term"],
+                "standardOntologyUris": [sampled_condition["sctid"]],
+            },
+            "otherRelevantDifferentials": [],
+            "impossibleConditions": [],
+            "correctCondition": {
+                "id": sampled_condition["id"],
+                "name": sampled_condition["term"],
+                "standardOntologyUris": [sampled_condition["sctid"]],
+            },
+        },
```

Comment:
We need to store the metadata for the case. A way of doing it would be to push it here alongside `caseData` and `valuesToPredict`. Otherwise we would need to change the `Case` data model (see comment below).

Reply:
Thanks @maccinza - good catch! Have changed this to add the metadata to the case alongside `caseData` and `valuesToPredict` as suggested